src/test/regress/expected/citus_local_tables_mx.out

```diff
@@ -237,6 +237,159 @@ SELECT stxname FROM pg_statistic_ext ORDER BY stxname;
+-- undistribute table
+-- this one should error out since we don't set the cascade option as true
+SELECT undistribute_table('cas_par2');
+ERROR:  cannot complete operation because table cas_par2 has a foreign key
+-- this one should work
+SELECT undistribute_table('cas_par2', cascade_via_foreign_keys=>true);
+NOTICE:  converting the partitions of citus_local_tables_mx.cas_par2
+NOTICE:  creating a new table for citus_local_tables_mx.cas_par2
+NOTICE:  dropping the old citus_local_tables_mx.cas_par2
+NOTICE:  renaming the new table to citus_local_tables_mx.cas_par2
+NOTICE:  creating a new table for citus_local_tables_mx.cas_1
+NOTICE:  moving the data of citus_local_tables_mx.cas_1
+NOTICE:  dropping the old citus_local_tables_mx.cas_1
+NOTICE:  renaming the new table to citus_local_tables_mx.cas_1
+NOTICE:  converting the partitions of citus_local_tables_mx.cas_par
+NOTICE:  creating a new table for citus_local_tables_mx.cas_par
+NOTICE:  dropping the old citus_local_tables_mx.cas_par
+NOTICE:  renaming the new table to citus_local_tables_mx.cas_par
+NOTICE:  creating a new table for citus_local_tables_mx.cas_par_1
+NOTICE:  moving the data of citus_local_tables_mx.cas_par_1
+NOTICE:  dropping the old citus_local_tables_mx.cas_par_1
+NOTICE:  renaming the new table to citus_local_tables_mx.cas_par_1
+NOTICE:  creating a new table for citus_local_tables_mx.cas_par_2
+NOTICE:  moving the data of citus_local_tables_mx.cas_par_2
+NOTICE:  dropping the old citus_local_tables_mx.cas_par_2
+NOTICE:  renaming the new table to citus_local_tables_mx.cas_par_2
+NOTICE:  creating a new table for citus_local_tables_mx.cas_par2_1
+NOTICE:  moving the data of citus_local_tables_mx.cas_par2_1
+NOTICE:  dropping the old citus_local_tables_mx.cas_par2_1
+NOTICE:  renaming the new table to citus_local_tables_mx.cas_par2_1
+NOTICE:  creating a new table for citus_local_tables_mx.cas_par2_2
+NOTICE:  moving the data of citus_local_tables_mx.cas_par2_2
+NOTICE:  dropping the old citus_local_tables_mx.cas_par2_2
+NOTICE:  renaming the new table to citus_local_tables_mx.cas_par2_2
+NOTICE:  converting the partitions of citus_local_tables_mx.cas_par3
+NOTICE:  creating a new table for citus_local_tables_mx.cas_par3
+NOTICE:  dropping the old citus_local_tables_mx.cas_par3
+NOTICE:  renaming the new table to citus_local_tables_mx.cas_par3
+NOTICE:  creating a new table for citus_local_tables_mx.cas_par3_1
+NOTICE:  moving the data of citus_local_tables_mx.cas_par3_1
+NOTICE:  dropping the old citus_local_tables_mx.cas_par3_1
+NOTICE:  renaming the new table to citus_local_tables_mx.cas_par3_1
+NOTICE:  creating a new table for citus_local_tables_mx.cas_par3_2
+NOTICE:  moving the data of citus_local_tables_mx.cas_par3_2
+NOTICE:  dropping the old citus_local_tables_mx.cas_par3_2
+NOTICE:  renaming the new table to citus_local_tables_mx.cas_par3_2
+ undistribute_table
+---------------------------------------------------------------------
+
+(1 row)
```

Comment:
after undistribute completes, can we print some records from catalog tables to verify that we preserve partitioning hierarchy ?

e.g.:
```sql
select inhrelid::regclass from pg_inherits where inhparent='cas_par'::regclass order by 1;
┌───────────┐
│ inhrelid  │
├───────────┤
│ cas_par_1 │
│ cas_par_2 │
└───────────┘
(2 rows)
```

Can you also go over other tests in this file and in other test files that we add with this pr for that ?
While doing that, please consider **not only** `undistribute_table`, but also explicit calls made to `create_citus_local_table`, and implicit local table creation / auto-undistribution cases.

Reply:
Can we also run some queries against pg_dist_partition before / after such operations to verify that child & shell tables are citus local tables / postgres local tables after completing such operations ?

Reply:
> e.g.:
> 
> ```sql
> select inhrelid::regclass from pg_inherits where inhparent='cas_par'::regclass order by 1;
> ┌───────────┐
> │ inhrelid  │
> ├───────────┤
> │ cas_par_1 │
> │ cas_par_2 │
> └───────────┘
> (2 rows)
> ```

I think we should also do that for parent "shard rel". We don't need to know the exact shard id when querying pg_inherits for parent shard, but we can filter by using regex or such,
```sql
select inhrelid::regclass from pg_inherits where (select inhparent::regclass::text) ~ '^cas_par_\d{6}$';
```